drodrpg/DRODLib/Character.cpp

```diff
@@ -2490,6 +2497,7 @@ void CCharacter::Process(
 			break;
 			case CCharacterCommand::CC_WaitForPlayerToTouchMe:
 			{
+				if (bRoomBeingDisplayedOnly) return;
```

Comment:
I'm thinking that instead of a `return` here, we should probably execute `STOP_COMMAND` instead. This is so that if this command is being used as part of an "if" condition, it will return false and continue script execution in the else block, etc.

Reply:
Good shout on checking for ifs. However in the case of wait for player to touch me, and many of the other commands that do things with the player, they should really all fail all the time if you're room previewing, one of the major issues that lead to this bug is that the player maintains his X and Y coordinates when room previewing so many characters were now thinking the player was touching them when the player wasn't even in the room.

If we're checking a condition of "is the player doing X" and we're not even looking at the room the player is in, I think the answer should always be "no".

Reply:
That's right. So, to get the script flow to process cleanly in all cases, we want to do this:
`player.wX == this->wX && player.wY == this->wY && !bRoomBeingDisplayedOnly) ...`